app/src/main/kotlin/at/bitfire/davdroid/ui/AppTheme.kt

```diff
@@ -30,9 +31,9 @@ fun AppTheme(
     val view = LocalView.current
     if (!view.isInEditMode) {
         SideEffect {
-            val window = (view.context as Activity).window
-            window.statusBarColor = colorScheme.primary.toArgb()
-            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = darkTheme
+            SystemBarStyle.auto(Color.TRANSPARENT, Color.TRANSPARENT) { darkTheme }.let {
+                (view.context as? AppCompatActivity)?.enableEdgeToEdge(it, it)
```

Comment:
This looks unecessarily convoluted to me. Can we make it more clear with named arguments or simply use a val instead of `let`?